kymatio/scattering2d/tests/test_torch_scattering2d.py

```diff
@@ -0,0 +1,376 @@
+""" This script will test the submodules used by the scattering module"""
+
+import os
+import numpy as np
+import torch
+import pytest
+from kymatio.scattering2d import Scattering2D
+from torch.autograd import gradcheck
+from kymatio.backend.fake_backend import backend as fake_backend
+
+backends = []
+
+try:
+    if torch.cuda.is_available():
+        from skcuda import cublas
+        import cupy
+        from kymatio.scattering2d.backend.torch_skcuda_backend import backend
+        backends.append(backend)
+except:
+    pass
+
+
+from kymatio.scattering2d.backend.torch_backend import backend
+backends.append(backend)
+
+if torch.cuda.is_available():
+    devices = ['cuda', 'cpu']
+else:
+    devices = ['cpu']
+
+
+# Checked the modulus
+class TestModulus:
+    @pytest.mark.parametrize("device", devices)
+    @pytest.mark.parametrize("backend", backends)
+    def test_Modulus(self, device, backend):
+        modulus = backend.modulus
+        x = torch.rand(100, 10, 4, 2).to(device)
+
+        if device == 'cpu' and backend.name == 'torch_skcuda':
```

Comment:
"Again, we shouldn't be testing for this here. Either we parametrize over backend–device tuples or we have a quick pytest.skip at the beginning of the test for this situation." Comment by @janden .

Reply:
Hmmm why not testing it raises an error? Sorry I'm confused, because it helps to show an error is well handled

Reply:
Good point. Maybe we could open an issue?

Reply:
Will do. This and the parametrizing over `backend` and `device` in a sensible way could simplify these tests a lot.